packages/core/src/amplitudeCore.ts

```diff
@@ -0,0 +1,17 @@
+import { AnalyticsConnectorImpl } from './analyticsConnector';
+import { IdentityStoreImpl } from './identityStore';
+import { safeGlobal } from './util/global';
+
+safeGlobal['amplitudeCoreInstances'] = {};
+
+export class AmplitudeCore {
+  public readonly identityStore = new IdentityStoreImpl();
+  public readonly analyticsConnector = new AnalyticsConnectorImpl();
+
+  static getInstance(instanceName: string): AmplitudeCore {
+    if (!safeGlobal['amplitudeCoreInstances'][instanceName]) {
```

Comment:
Is this to resolve earlier issue mentioned ?

Reply:
Yes, even if the same code is copied and used separately, the core instance is always a truly global singleton.